### PR TITLE
Cknieling/add report support

### PR DESCRIPTION
**Pull request by chr11115**
- add basic support for reports

### CODE REVIEW

**Review by bneukom — CHANGES_REQUESTED** [review]
Compile issue and some minor comments.

File: VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Public/ReportAttribute.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+#pragma once
+
+#include "prt/AttributeMap.h"
+
+#include "VitruvioTypes.h"
+
+#include "Misc/Variant.h"
+
+struct FReportAttribute {
```
[the remark this answers is not in the document]
Comment (chr11115) [reply VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Public/ReportAttribute.h:24]: renamed to FReport

File: VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Public/ReportAttribute.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	FString Name;
```
[the remark this answers is not in the document]
Comment (chr11115) [reply VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Public/ReportAttribute.h:25]: Renamed to Key

File: VitruvioHost/Plugins/Vitruvio/Extras/UnrealGeometryEncoder/Private/Codec/Encoder/UnrealGeometryEncoder.cpp [hunk 3]
```diff
@@ -607,6 +615,14 @@ void UnrealGeometryEncoder::encode(prtx::GenerateContext& context, size_t initia
 	prtx::EncodePreparator::InstanceVector instances;
 	encPrep->fetchFinalizedInstances(instances, PREP_FLAGS);
 	convertGeometry(initialShape, instances, cb);
+
+	if (emitAttrs) {
+		const prtx::ReportsPtr& reports = reportsCollector->getReports();
+		if (reports != nullptr) {
```
[the remark this answers is not in the document]
Comment (chr11115) [reply VitruvioHost/Plugins/Vitruvio/Extras/UnrealGeometryEncoder/Private/Codec/Encoder/UnrealGeometryEncoder.cpp:621]: fixed.

File: VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Public/VitruvioComponent.h [hunk 4]
```diff
@@ -234,6 +234,10 @@ class VITRUVIO_API UVitruvioComponent : public UActorComponent
 	UPROPERTY(EditAnywhere, BlueprintReadOnly, DisplayName = "Random Seed", Category = "Vitruvio", meta = (AllowPrivateAccess = "true"))
 	int32 RandomSeed;
 
+	/** CGA Reports from generation. */
+	UPROPERTY(EditAnywhere, DisplayName = "Reports", Category = "Vitruvio")
```
[the remark this answers is not in the document]
Comment (chr11115) [reply VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Public/VitruvioComponent.h:238]: removed Uproperty tag for now. 
The blueprint API will be decided later

File: VitruvioHost/Plugins/Vitruvio/Extras/UnrealGeometryEncoder/Public/Codec/Encoder/IUnrealCallbacks.h [hunk 5]
```diff
@@ -51,12 +52,14 @@ class IUnrealCallbacks : public prt::Callbacks
...
+	 *                    the call to addInstance and addReport
 	 * @param transform the transformation matrix of this instance
 	 * @param instanceMaterial override materials for this instance
 	 * @param numInstanceMaterials number of instance material overrides. Is either 0 or is equal to the number
 	 *                             of materials of the original mesh (by prototypeId)
 	 */
 	virtual void addInstance(int32_t prototypeId, const double* transform, const prt::AttributeMap** instanceMaterial,
 							 size_t numInstanceMaterials) = 0;
+	
+	virtual void addReport(const prtx::PRTUtils::AttributeMapPtr reports) = 0;
```
[the remark this answers is not in the document]
Comment (chr11115) [reply VitruvioHost/Plugins/Vitruvio/Extras/UnrealGeometryEncoder/Public/Codec/Encoder/IUnrealCallbacks.h:64]: passing normal pointer now.

**Pull request merged**